provider/lxd/environ.go

```diff
@@ -456,3 +460,27 @@ func (env *environ) AssignLXDProfiles(instID string, profilesNames []string, pro
 	}
 	return report(nil)
 }
+
+// DetectSeries is a no-op for lxd, must return an empty string.
+func (env *environ) DetectSeries() (string, error) {
+	return "", nil
+}
+
+// DetectHardware returns the hardware characteristics for the controller for
+// this environment. This method is part of the environs.HardwareCharacteristicsDetector
+// interface. On an LXD cloud, it must first check if it is a localhost cloud,
+// in that case, we only fill the Arch.
+func (env *environ) DetectHardware() (*instance.HardwareCharacteristics, error) {
+	// The returned error is willingly ignored, because this should not
+	// break bootstrapping (we detect hardware before bootstrapping),
+	// instead, bootstrapping should fallback to default hardware arch.
+	isLocalhostCloud, _ := env.cloud.IsLocalhostCloud()
```

Comment:
Just move the logic from the above cloudspec method to here.

Reply:
done